src/main/java/pl/javastart/task/components/OverclockableComponent.java

```diff
@@ -0,0 +1,47 @@
+package pl.javastart.task.components;
+
+import pl.javastart.task.exception.IllegalClockSpeedException;
+import java.util.InputMismatchException;
+
+abstract class OverclockableComponent extends Component implements Overclockable {
+    private final Config config;
+
+    public OverclockableComponent(String model, String manufacturer, String serialNo,
+                                  int temperatureIncrease, int clockSpeed, int temperature, int maxTemperature) {
+        super(model, manufacturer, serialNo);
+        config = new Config(temperatureIncrease, clockSpeed, temperature, maxTemperature);
+    }
+
+    @Override
+    public void overclock(int targetClockSpeed) throws InputMismatchException, IllegalClockSpeedException {
+        if (targetClockSpeed < config.getClockSpeed()) {
+            throw new IllegalClockSpeedException("\nThis is supposed to be an overclocking, not downclocking. "
+                    + targetClockSpeed + " is therefore an incorrect target clock speed.\n");
+        }
+
+        int targetTemperature = config.getTemperature() + (targetClockSpeed - config.getClockSpeed()) / 100 * config.getTemperatureIncrease();
+
+        if (targetTemperature > config.getMaxTemperature()) {
+            throw new IllegalClockSpeedException("\n" + targetClockSpeed + " clock speed would burn the component\n");
```

Comment:
Chciałbym, żebyś jeszcze uspójnił język aplikacji. Z jednej strony wyświetlasz komunikaty w języku polskim, a z drugiej masz komunikaty z wyjątków po angielsku. Zamiast ustawiać tutaj message, może lepiej rzucić np. taki wyjątek:
```
throw new IllegalClockSpeedException(targetClockSpeed, targetTemperature);
```
i na ich podstawie już w bloku catch zbudować sobie wiadomość w języku polskim

Reply:
Przy użyciu tu języka angielskiego sugerowałem się zaleceniem z Waszego kursu, by treści wyjątków były zachowane właśnie w języku angielskim ze względu na ich użyteczność dla programistów. No ale czy dobrze zgaduję, że zalecenie to tyczyło się jedynie wyjątków nieobsłużonych, które w stack trace'ie "wypływają" aż do metody main? A w sytuacji jak ta w rozpatrywanym tu programie, gdzie złapanie takiego wyjątku posłużyć ma do wygenerowania komunikatu dedykowanego dla usera, komunikat powinien być jednak w języku polskim?

Reply:
Tak, trzeba rozróżnić treść wyjątków, która używana jest na potrzeby logowania i debuggowania, od tego co widzi ostatecznie użytkownik w konsoli, czy na stronie internetowej